src/posts/components/CreatePostButton/CreatePostForm.tsx

```diff
@@ -0,0 +1,305 @@
+              noItemMessage="No board found"
+              searchInputPlaceholder="Search board..."
+            />
+          </DropdownMenu>
+        </div>
+        <button className="p-1 bg-transparent hover:bg-muted rounded-md" onClick={onClose}>
+          <X className="w-4.5 h-4.5 cursor-pointer text-muted-foreground" />
+        </button>
+      </div>
+
+      <div className="relative mt-1 z-[50] px-1">
+        <input
+          id="create-post-title"
+          autoFocus
+          className="text-base font-medium bg-transparent border-0 w-full text-foreground sm:text-lg focus:outline-none focus:ring-0 px-4 py-2"
+          placeholder="Title of your post"
+          value={title}
+          onChange={(e) => {
+            if (e.target.value?.length > 2 && isSubmitted) {
+              setIsSubmitted(false);
+            }
+            setTitle(e.target.value);
+          }}
+        />
+      </div>
+
+      {isSubmitted && isInvalid ? (
+        <div className="px-4">
+          <p className="text-sm text-red-500/80 mt-0.5">Title must be longer that 2 characters.</p>
+        </div>
+      ) : null}
+
+      <div className="px-1 mb-1">
+        <ProseMirrorEditor placeholder="Post description..." onChange={setDescription} />
+      </div>
+
+      {selectedBoard?.label === 'Bugs' ? (
+        <>
+          <div className="relative z-[50] py-3.5 px-4 bg-popover border-t border-accent">
+            <p className="pr-20 text-sm font-medium text-gray-400 dark:text-foreground">
+              What is experiencing an issue?
+            </p>
+            <div className="flex flex-wrap mt-3 gap-2.5">
+              {badgeOptions.map(({ value, label }) => (
+                <BugBadge
+                  key={value}
+                  value={value}
+                  label={label}
+                  onSelect={(val) =>
+                    setSelectedBugSources((prev) =>
+                      prev?.includes(val) ? prev.filter((src) => src !== val) : [...(prev || []), val]
+                    )
+                  }
+                  isSelected={(selectedBugSources || [])?.some((src) => src === value)}
+                />
+              ))}
+            </div>
+          </div>
+
+          <div className="relative z-[50] px-4 bg-popover border-t border-accent">
+            <p className="pt-3.5 pointer-events-none absolute text-xs font-medium text-gray-400 dark:text-foreground">
+              Environment?
+            </p>
+            <textarea
+              id="environment"
+              className="sm:text-[15px] min-h-[70px] max-h-[170px] field-sizing-content w-full resize-none pr-32 dark:bg-transparent custom-scrollbar border-transparent dark:border-transparent bg-transparent pt-[36px] pb-3 rounded-none border-x-0 dark:text-white focus-within:outline-none focus-within:ring-0 dark:focus-within:outline-none"
+              placeholder="Which browser and device?"
+            ></textarea>
```

Comment:
is there a reason why we have a textarea tag with custom style like this instead of using the shadcn component?